File: implementations/ruby/stat_process.rb
class StatProcess
  @name
  @version
  @description
  @maintainer
  @email
  @website
  @repeatability

  def initialize(name)
    throw :typeError unless name.is_a?(String)
    @name = name
  end

  def name=(name)
    throw :typeError unless name.is_a?(String)
    @name = name
  end

  def name
    @name
  end

  def version=(version)
    throw :typeError unless version.is_a?(String)
    @version = version
  end

  def version
    @version
  end

  def description=(description)
    throw :typeError unless description.is_a?(String)
    @description = description
  end

  def description
    @description
  end

  def maintainer=(maintainer)
    throw :typeError unless maintainer.is_a?(String)
    @maintainer = maintainer
  end

  def maintainer
    @maintainer
  end

  def email=(email)
    throw :typeError unless email.is_a?(String)
    @email = email
  end

  def email
    @email
  end

  def website=(website)
    throw :typeError unless website.is_a?(String)
    @website = website
  end

  def website
    @website
  end

  def repeatability=(repeatability)
    throw :typeError unless Repeatability.all.include?(repeatability)
    @repeatability = repeatability
  end

  def repeatability
    @repeatability
  end
end

File: implementations/php/src/StructuredAcceptanceTest/StatProcess.php
<?php

namespace StructuredAcceptanceTest;

class StatProcess {

    /* Required — Identification of the acceptance test process */
    public $name;

    /* Optional — The version of the acceptance test process */
    public $version;

    /* Optional — A brief explanation of the acceptance test process */
    public $description;

    /* Optional — The name of a person responsible for the acceptance test process */
    public $maintainer;

    /* Optional — Contact email address for the maintainer */
    public $email;

    /* Optional — Contact website for the acceptance test process */
    public $website;

    /* Optional — A guarantee of whether the same validation output can be expected in future validation, see notes below */
    // Volatile — Findings MAY change when repeating validation on the identical targets. Example: a web link checker, or package manager version checker. This is the implicit default if not specified.
    // Repeatable — Findings MUST be identical if the program is run again with the same inputs. ("Inputs" is not specified by this standard.)
    // Associative — Findings for targets [a, b] MUST equal the union of findings for [a] and [b] -- in other words, if only one file in changed, only that file need be tested.
    public $repeatability;
}
